File: 2-add-two-numbers/add-two-numbers.py
class Solution:
    def addTwoNumbers(self, a, b):
        d = ListNode(0)
        c = d
        s = 0
        
        while a or b or s:
            if a:
                s += a.val
                a = a.next
            if b:
                s += b.val
                b = b.next
            
            c.next = ListNode(s % 10)
            c = c.next
            s //= 10
        
        return d.next

File: 2-add-two-numbers/add-two-numbers.java
class Solution {
    public ListNode addTwoNumbers(ListNode a, ListNode b) {
        ListNode d = new ListNode(0);
        ListNode c = d;
        int s = 0;
        
        while (a != null || b != null || s != 0) {
            if (a != null) {
                s += a.val;
                a = a.next;
            }
            if (b != null) {
                s += b.val;
                b = b.next;
            }
            
            c.next = new ListNode(s % 10);
            c = c.next;
            s /= 10;
        }
        return d.next;
    }
}
